ts/src/pro/poloniex.ts

```diff
@@ -0,0 +1,970 @@
+        //        "price": "47112.1",
+        //        "tradeQty": "0",
+        //        "tradePrice": "0",
+        //        "tradeId": "0",
+        //        "ts": 1648708187469
+        //    }
+        //
+        const timestamp = this.safeInteger (trade, 'tradeTime');
+        const marketId = this.safeString (trade, 'symbol');
+        return this.safeTrade ({
+            'info': trade,
+            'id': this.safeString (trade, 'tradeId'),
+            'symbol': this.safeSymbol (marketId, market),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'order': this.safeString (trade, 'orderId'),
+            'type': this.safeStringLower (trade, 'type'),
+            'side': this.safeString (trade, 'side'),
+            'takerOrMaker': this.safeString (trade, 'matchRole'),
+            'price': this.safeString (trade, 'price'),
+            'amount': this.safeString (trade, 'tradeAmount'),
+            'cost': undefined,
+            'fee': {
+                'rate': undefined,
+                'cost': this.safeString (trade, 'tradeFee'),
+                'currency': this.safeString (trade, 'feeCurrency'),
+            },
+        }, market);
+    }
+
+    handleOrder (client, message) {
+        //
+        // Order is created
+        //
+        //    {
+        //        channel: 'orders',
+        //        data: [
+        //            {
+        //                "symbol": "BTC_USDT",
+        //                "type": "LIMIT",
+        //                "quantity": "1",
+        //                "orderId": "32471407854219264",
+        //                "tradeFee": "0",
+        //                "clientOrderId": "",
+        //                "accountType": "SPOT",
+        //                "feeCurrency": "",
+        //                "eventType": "place",
+        //                "source": "API",
+        //                "side": "BUY",
+        //                "filledQuantity": "0",
+        //                "filledAmount": "0",
+        //                "matchRole": "MAKER",
+        //                "state": "NEW",
+        //                "tradeTime": 0,
+        //                "tradeAmount": "0",
+        //                "orderAmount": "0",
+        //                "createTime": 1648708186922,
+        //                "price": "47112.1",
+        //                "tradeQty": "0",
+        //                "tradePrice": "0",
+        //                "tradeId": "0",
+        //                "ts": 1648708187469
+        //            }
+        //        ]
+        //    }
+        //
+        const data = this.safeValue (message, 'data');
+        let orders = this.orders;
+        if (orders === undefined) {
+            const limit = this.safeInteger (this.options, 'ordersLimit');
+            orders = new ArrayCacheBySymbolById (limit);
+            this.orders = orders;
+        }
+        const order = this.safeValue (data, 0);
```

Comment:
Can it happen that more than one order can be passed in the array? To be safe I'd put this in a for loop